### ISSUE TITLE
UnicodeDecodeError in posixpath on ipython==5.1.0

### ISSUE THREAD

**Issue by lefterisnik** [jazzband/pip-tools#434]
I have run `pip-compile` with python 2.7 and it failed with the above error:

```
pip-compile -o base.txt base.in
No handlers could be found for logger "pip.req.req_install"
Traceback (most recent call last):
  File "/usr/local/bin/pip-compile", line 11, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/piptools/scripts/compile.py", line 185, in cli
    results = resolver.resolve()
  File "/usr/local/lib/python2.7/dist-packages/piptools/resolver.py", line 109, in resolve
    has_changed, best_matches = self._resolve_one_round()
  File "/usr/local/lib/python2.7/dist-packages/piptools/resolver.py", line 191, in _resolve_one_round
    for best_match in best_matches
  File "/usr/local/lib/python2.7/dist-packages/piptools/resolver.py", line 192, in <genexpr>
    for dep in self._iter_dependencies(best_match))
  File "/usr/local/lib/python2.7/dist-packages/piptools/resolver.py", line 261, in _iter_dependencies
    dependencies = self.repository.get_dependencies(ireq)
  File "/usr/local/lib/python2.7/dist-packages/piptools/repositories/local.py", line 68, in get_dependencies
    return self.repository.get_dependencies(ireq)
  File "/usr/local/lib/python2.7/dist-packages/piptools/repositories/pypi.py", line 135, in get_dependencies
    dependencies = reqset._prepare_file(self.finder, ireq)
  File "/usr/local/lib/python2.7/dist-packages/pip/req/req_set.py", line 620, in _prepare_file
    session=self.session, hashes=hashes)
  File "/usr/local/lib/python2.7/dist-packages/pip/download.py", line 821, in unpack_url
    hashes=hashes
  File "/usr/local/lib/python2.7/dist-packages/pip/download.py", line 663, in unpack_http_url
    unpack_file(from_path, location, content_type, link)
  File "/usr/local/lib/python2.7/dist-packages/pip/utils/__init__.py", line 599, in unpack_file
    flatten=not filename.endswith('.whl')
  File "/usr/local/lib/python2.7/dist-packages/pip/utils/__init__.py", line 492, in unzip_file
    fn = os.path.join(location, fn)
  File "/usr/lib/python2.7/posixpath.py", line 80, in join
    path += '/' + b
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc6 in position 27: ordinal not in range(128)
```

the base.in file looks like:

```
...
django-bootstrap3==7.0.1
Django==1.10.5
djangorestframework-jwt==1.8.0
djangorestframework==3.4.6
icalendar==3.10
ipython==5.1.0
...
```
Any ideas?

**Comment by byjott** [comment]
This looks very much like a duplicate of #433

**Comment by mhluongo** [comment]
@byjott It does look similar, but your fix for #433 isn't working (at least in our fork)

**Comment by vphilippon** [comment]
Should be fixed by #568, closing.
